lib/common/user_service.dart

```diff
@@ -75,67 +74,55 @@ class UserService {
-                      isLogin = true;
-                    }
-                  }).catchError((e) {
-                    displayError(context, e);
-                  });
-                },
+              content: Column(
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  Text(maskPhoneNumber(phoneNumber.toString())),
+                  OtpInputWidget(
+                    onOtpEntered: (otp) async {
+                      FirebaseAuth auth = FirebaseAuth.instance;
+                      String smsCode = otp;
+                      _credential = PhoneAuthProvider.credential(verificationId: verificationId, smsCode: smsCode);
+                      auth.signInWithCredential(_credential).then((result) async {
+                        Navigator.pushReplacement(context, MaterialPageRoute(builder: (context) => const HomePage()));
+                        bool exists = await userExists(result.user!.uid);
+                        if (!exists) {
+                          UserData newUser = UserData(
+                            userId: result.user!.uid,
+                            userName: 'guest',
+                            email: result.user!.email ?? '',
+                            phoneNumber: phoneNumber,
+                            profilePictureUrl: Constants.profileAvatarUrl,
+                            userType: UserType.user.name,
+                          );
+                          await createUser(newUser);
+                          showSnackbar(context, "registration_massage".i18n());
+                        } else if (exists) {
+                          showSnackbar(context, "login_massage".i18n());
+                        }
+                      }).catchError((e) {
+                        displayError(context, e);
+                      });
+                    },
+                  ),
+                ],
               ),
             );
           },
         );
       },
-      codeAutoRetrievalTimeout: (String verificationId) {
-        isLogin ? null : context.pop();
-        showDialog(
-          context: context,
-          builder: (BuildContext context) {
-            return AlertDialog(
-              title: Text("time_out".i18n()),
-              content: Text("time_out_massage".i18n()),
-              actions: <Widget>[
-                TextButton(
-                  child: Text("ok".i18n()),
-                  onPressed: () {
-                    context.pop();
-                  },
-                ),
-              ],
-            );
-          },
-        );
-      },
+      codeAutoRetrievalTimeout: (String verificationId) {},
     );
   }
 
   /// Checks if a user document exists for the given ID
   Future<bool> userExists(String userId) async {
     DocumentSnapshot doc = await fireStore.collection('users').doc(userId).get();
-    return doc.exists;
+    if (doc.exists) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
this can be simplified to 
```suggestion
    return await fireStore.collection('users').doc(userId).get().exists;
```

please verify.

Reply:
i tried to use this code but i faced this issue

The getter 'exists' isn't defined for the type 'Future<DocumentSnapshot<Map<String, dynamic>>>'.
Try importing the library that defines 'exists', correcting the name to the name of an existing getter, or defining a getter or field named 'exists'.